crates/exex/exex/src/backfill.rs

```diff
@@ -185,41 +185,70 @@ where
         let chain = Chain::new(blocks, executor.finalize(), None);
         Ok(chain)
     }
+
+    /// Returns an iterator that executes each block in the range separately and yields the
+    /// resulting [`Block`] and [`ExecutionOutcome`].
+    pub fn into_single_blocks(
+        &self,
+    ) -> impl Iterator<Item = Result<(Block, ExecutionOutcome), BlockExecutionError>> + '_ {
+        self.range.clone().map(|block_number| {
```

Comment:
`into` methods consume
```suggestion
        self,
    ) -> impl Iterator<Item = Result<(Block, ExecutionOutcome), BlockExecutionError>> + '_ {
        self.range.map(|block_number| {
```